tpm2/tpm2.go

```diff
@@ -64,70 +68,89 @@ func encodeTPMLPCRSelection(sel PCRSelection) ([]byte, error) {
 	// For example, selecting PCRs 3 and 9 looks like:
 	// size(3)  mask     mask     mask
 	// 00000011 00000000 00000001 00000100
-	ts := tpmsPCRSelection{
-		Hash: sel.Hash,
-		Size: sizeOfPCRSelect,
-		PCRs: make(tpmutil.RawBytes, sizeOfPCRSelect),
-	}
-	// sel.PCRs parameter is indexes of PCRs, convert that to set bits.
-	for _, n := range sel.PCRs {
-		byteNum := n / 8
-		bytePos := byte(1 << byte(n%8))
-		ts.PCRs[byteNum] |= bytePos
+	var retBytes []byte
+	for i := 0; i < len(s); i++ {
+		ts := tpmsPCRSelection{
+			Hash: s[i].Hash,
+			Size: sizeOfPCRSelect,
+			PCRs: make(tpmutil.RawBytes, sizeOfPCRSelect),
+		}
+		// s[i].PCRs parameter is indexes of PCRs, convert that to set bits.
+		for _, n := range s[i].PCRs {
+			byteNum := n / 8
+			bytePos := byte(1 << byte(n%8))
+			ts.PCRs[byteNum] |= bytePos
+		}
+		tmpBytes, err := tpmutil.Pack(uint32(1), ts)
```

Comment:
`TPML_PCR_Selection` encoding should be:
`[len(sel), sel[0], sel[1], ...]`
here you're producing:
`[1, sel[0], 1, sel[1], ...]`

(the hardcoded `uint32(1)` in the old implementation was `len(sel)` because I was too lazy to handle a list)